bugbug/models/performancebug.py

```diff
@@ -0,0 +1,150 @@
+    whiteboard = bug_data.get("whiteboard", "").lower()
+    cf_performance = bug_data.get("cf_performance", "")
+
+    performance_keywords = ("perf", "topperf", "main-thread-io")
+
+    whiteboard_performance_text = (
+        "fxperf",
+        "fxperfsize",
+        "snappy",
+        "pdfjs-c-performance",
+        "pdfjs-performance",
+        "sp3",
+    )
+
+    if any(
+        keyword.startswith(performance_keyword)
+        for keyword in keywords
+        for performance_keyword in performance_keywords
+    ):
+        return True
+
+    if any(f"[{text}" in whiteboard for text in whiteboard_performance_text):
+        return True
+
+    if cf_performance and cf_performance not in ("---", "?"):
+        return True
+
+    return False
+
+
+class PerformanceBugModel(BugModel):
+    def __init__(self, lemmatization=False):
+        BugModel.__init__(self, lemmatization)
+
+        self.calculate_importance = False
+
+        feature_extractors = [
+            bug_features.HasSTR(),
+            bug_features.Severity(),
+            bug_features.Keywords(),
+            bug_features.IsCoverityIssue(),
+            bug_features.HasCrashSignature(),
+            bug_features.HasURL(),
+            bug_features.HasW3CURL(),
+            bug_features.HasGithubURL(),
+            bug_features.Whiteboard(),
+            bug_features.Patches(),
+            bug_features.Landings(),
+            bug_features.BlockedBugsNumber(),
+            bug_features.EverAffected(),
+            bug_features.AffectedThenUnaffected(),
+            bug_features.Product(),
+            bug_features.Component(),
+        ]
+
+        cleanup_functions = [
+            feature_cleanup.fileref(),
+            feature_cleanup.url(),
+            feature_cleanup.synonyms(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                (
+                    "bug_extractor",
+                    bug_features.BugExtractor(feature_extractors, cleanup_functions),
+                ),
+            ]
+        )
+
+        self.clf = ImblearnPipeline(
+            [
+                (
+                    "union",
+                    ColumnTransformer(
+                        [
+                            ("data", DictVectorizer(), "data"),
+                            ("title", self.text_vectorizer(min_df=0.0001), "title"),
+                            (
+                                "first_comment",
+                                self.text_vectorizer(min_df=0.001),
+                                "first_comment",
+                            ),
+                            (
+                                "comments",
+                                self.text_vectorizer(min_df=0.0001),
+                                "comments",
+                            ),
+                        ]
+                    ),
+                ),
+                ("sampler", BorderlineSMOTE(random_state=0)),
```

Comment:
Would under-sampling be more effective than over-sampling in this case?

Reply:
> Would under-sampling be more effective than over-sampling in this case?

Since performance bugs are fewer than non-performance bugs, I believe that despite being more resource-intensive, over-sampling would be more effective. Additionally, it could help prevent the discarding of valuable bug data. We may need to assess the impact of each on performance to be completely certain.

Reply:
The performance metrics indicate that [over-sampling](https://github.com/mozilla/bugbug/pull/3895#issuecomment-1848993947), appears to outperform [under-sampling](https://github.com/mozilla/bugbug/pull/3895#issuecomment-1849824872).

- Over-sampling:
```
Accuracy: f0.9983842479309637 (+/- 0.00035056001978153533)
Precision: f0.9830817331894579 (+/- 0.01008477889026195)
Recall: f0.9499860155507076 (+/- 0.01479548496312242)
```

- Under-sampling:
```
Accuracy: f0.9867026927734885 (+/- 0.001089275983816713)
Precision: f0.6518237660231335 (+/- 0.02116194994745171)
Recall: f0.9745702778510296 (+/- 0.012108931726627428)
```